powertools-parameters/src/test/java/software/amazon/lambda/powertools/parameters/SSMProviderTest.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package software.amazon.lambda.powertools.parameters;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import software.amazon.awssdk.services.ssm.SsmClient;
+import software.amazon.awssdk.services.ssm.model.*;
+import software.amazon.lambda.powertools.parameters.transform.Transformer;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.openMocks;
+
+public class SSMProviderTest {
```

Comment:
These test are again testing on a default method `getValue`which is still ok for unit testing, I would very much like to see end to end tests on these providers . Something like Having a instance from `ParamManagr` with a mocked sdk client and then validating different interactions etc based on cache value.

Reply:
getValue is actually what we want to test in those classes. This is the core method that retrieves the parameter with a mocked client, exactly what you mention. Not sure an integration test which does the same has any value. Maybe a real integration test with a real call to SSM would be interesting.